src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -227,6 +227,7 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.ConfigureOptions<AzureMonitorLoggingOptionsSetup>();
             services.AddHostingConfigOptions(configuration);
             services.ConfigureOptions<ExtensionRequirementOptionsSetup>();
+            services.Configure<HostMetricsProviderOptions>(configuration.GetSection("HostMetricsProvider")); // todo: use options setup?
```

Comment:
Small nit: I prefer this approach,

```suggestion
            services.AddOptions<HostMetricsProviderOptions>().BindConfiguration("HostMetricsProvider");
```

There is a slight difference between the two. What you have now is getting "host" configuration only. No "app" configuration. Additionally, the `BindConfiguration` will support config reloads.